gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/ProductDependencyMerger.java

```diff
@@ -92,6 +81,32 @@ public static ProductDependency merge(ProductDependency dep1, ProductDependency
         return result;
     }
 
+    private static String getMinimumVersion(
+            ProductDependency dep1, ProductDependency dep2, Optional<String> defaultMinimumVersion) {
+        if (defaultMinimumVersion.isPresent()
+                && (!OrderableSlsVersion.check(dep1.getMinimumVersion())
+                        || !OrderableSlsVersion.check(dep2.getMinimumVersion()))) {
+            return defaultMinimumVersion.get();
+        }
+
+        // This could be empty if both of the versions are dirty
+        Optional<OrderableSlsVersion> minimumVersionOrderable = Stream.of(dep1.parseMinimum(), dep2.parseMinimum())
```

Comment:
We can already return this version as the if-else checks are no longer necessary. If one of the versions is not parseable, we already return early in line `89`. Based on that, `minimumVersionOrderable` will never be empty.

Reply:
thats only true if default minimum is present, right?